src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -608,17 +628,14 @@ private Key CreateKey(string name, string secret)
             return _keyValueConverterFactory.WriteKey(key);
         }
 
-        internal static string GenerateSecret()
+        internal static string GenerateIdentifiableSecret(ulong seed)
         {
-            using (var rng = RandomNumberGenerator.Create())
-            {
-                byte[] data = new byte[40];
-                rng.GetBytes(data);
-                string secret = Convert.ToBase64String(data);
+            string secret = IdentifiableSecrets.GenerateIdentifiableKey(seed, 40, AzureFunctionsSignature);
 
-                // Replace pluses as they are problematic as URL values
-                return secret.Replace('+', 'a');
-            }
+            secret = secret.Replace('+', '-');
```

Comment:
>secret [](http://example.com/codeflow?start=12&length=6)

Previously, this code eliminated all '+' chars from the generated key, I believe because in at least one code path, for extensions, a key was returned as part of a URL. Generally, the '/' character is also replaced or eliminated if URL-friendly base64-encoded values are sought (and the '=' sign padding is also replaced.

What I've done here is simply to replace '+' with '-' and '/' with '_', a common convention for maintaining a base64-encoding with a complete character set.

This change, however, will change the alphabet of your keys moving forward. It will break any consumers who may have depended on the previous behavior to use functions like Convert.FromBase64String to convert a secret to a byte array.

The Azure Identity package provides an alternate which could now be used, the Base64UrlEncoder class. It would be important for this change to determine whether there are any code locations which s/be updated to use this API. Alternately, those code locations could simply find/replace back to the 'special' character set before calling Convert.

Reply:
As discussed, our keys need to continue to be URL friendly. They are passed as query params, but not in path segments which is why '/' isn't an issue. Also, we don't have any customers decoding these back to byte array - that's not a concern.

Reply:
Makes sense. Only the '+' character is a problem if tokens are only passed as query params. The trailing '=' padding chars shouldn't be a concern (as far as parameter parsing) if you're using the '&' as a separator.

Still, I think converting to a full 64-character (URL friendly) alphabet and eliding the trailing equal signs characters is a good change, since you've confirmed there shouldn't be anyone decoding to byte arrays (using standard base64) today.

One minor question I have is whether the absence of trailing padding might result in some incremental change in the # of false positive regex matches (that will get eliminated by the checksum validation). I plan to test this on a fairly large data set (a body of public OSS code) this week, will follow up here on results.